Support set map value by key and fix dependency

pingginp opened the pull request:
This PR fixes #36 and #39 Since the repo has not been maintained for a while I changed the readme and test, if it's possible to get those things merged, I can create a cleaner PR with different readme.

Improvements
 
- use latest gocql and fix constant naming #36 
- support set map value by key, previously only entire map is supported #39
- use docker for e2e test
- clean up the template so it won't have too many blank lines and block proper gofmt

Dependencies

- remove seelog, it is only used in generator cli, use standard log is enough
- use `pkg/errrors` to preserve context in cqlc runtime
- use logrus in cqlc runtime, could add a extra logger interface to allow using other library if really needed

Know issues

- integration test no longer works because they generate file in /tmp and can't use gocql under /vendor and will fail assertion due to type mismatch

And @0x6e6562 if you no longer have the time to maintain it, is it possible to update the repo doc and point to a maintained fork, thx.

pingginp committed:
[dep] Init glide https://github.com/pingginp/cqlc/issues/1

pingginp committed:
[dep] Fix gocql ColumnClusterKey etc.

- upstream https://github.com/relops/cqlc/issues/36

pingginp committed:
[gen] Format generated columns.go

- the template would result in extra space

pingginp committed:
[test] Bump go and cassandra version in Travis

- https://github.com/pingginp/cqlc/issues/3
- not sure if I can run ccm locally

pingginp committed:
[gen] Remove go-bindata and use template directly

- https://github.com/pingginp/cqlc/issues/5

pingginp committed:
[test] Install glide to $GOPATH/bin

- when use /usr/local/bin got permission denied ...

pingginp committed:
[test] Move project to right go path

- it should be using relops/cqlc, but default travis will be using
pingginp/cqlc, that's the problem of enable travis on fork

pingginp committed:
[test] Use C* 2.2.1

- it seems 3.11 is invalid for ccm

pingginp committed:
[test] Remove bindata rule from test deps

pingginp committed:
Fix Printf format error

- see https://travis-ci.org/pingginp/cqlc/builds/428905343

pingginp committed:
[test] go get github.com/gocql/cql for generator

- generator generates files in /tmp for testing, so need global gocql in
$GOPATH to work ...

pingginp committed:
[test] Only test cqlc, remove generator

- generator was comparing type, and due to use vendor and global go
path, there will be type mismatch https://travis-ci.org/pingginp/cqlc/builds/428908553

pingginp committed:
[gen] Add set map value by key #2

- only support `col[?] = ?`, value binding should work, because there
was `WHERE foo in (?,?,?)` so in `Prepare` cqlc would flat the slice
before passing to gocql
- https://github.com/pingginp/cqlc/issues/2
- [ ] TODO: need to test against a real database to see if it works

pingginp committed:
[test] Can't connect to C* using docker on Mac #6

pingginp committed:
Fix #6 set broadcast_addr and correct port

- in port mapping, was using 904 instead of 9042 ...

pingginp committed:
[gen] C*3 seems to be broken

pingginp committed:
[gen] Set Map value didn't work ...

- got "can not marshal []interface {} into map(varchar, varchar)"

pingginp committed:
[gen] Bind value for map didn't work #2

- when insert, use SetStringStringMap
- when update, SetStringStringMapValue didn't work ... got gocql: expected 3 values send got 2

pingginp committed:
[gen] Flatten binding in BuildStatement #2 works

- BuildStatement is called by Exec, Prepare is called by Fetch,
previously flatten is only applied in Prepare

pingginp committed:
[dep] Remove seelog add pkg/errors

pingginp committed:
[gen] Use separated type for map key value

- previous implementation is wrong, it blindly expand all value slice,
which would break AppendSlice, now we use struct KeyValue as binding
value and expand based on CollectionType and CollectionOperationType

pingginp committed:
[test] Run gocql and cqlc in parallel

- string map worked fine, not sure about slice though ...

pingginp committed:
[test] Fix #3 use docker for e2e test

- I don't know if old integration test still work though ....

pingginp committed:
[gen] Change generated header to canonical form

pingginp committed:
Merge pull request #4 from pingginp/reboot

Reboot

- Fix #1 we are now using glide
- Fix #2 support set map by key, the generated interface `cqlc/columns.go` is updated, the runtime `BuildStatement` is updated to flatten the `KeyValue` struct to a slice of length 2
- Fix #3 use docker-compose to test locally and on CI, the e2e works because we don't allow parallel and go will run them serial in the order they are defined. Also if the schema is changed, the test need to be run twice and the first one will fail (same as latex bib index), the generated package can only be pick up next time
- Fix #5 we now use template in go code using long text
- Fix #6 need to set broadcast_address to 127.0.0.1 and avoid typo in port mapping

pingginp committed:
[runtime] Use logrus instead of stdlog

- similar to #8

pingginp committed:
[gen] Replace log.Fatalf w/ errors.Errorf

- in generated table definitions, was using log.Fatalf, that function
returns error, better just return errors.Errorf

pingginp committed:
[runtime] Update version defined in runtime

- make update-ver

pingginp committed:
Merge pull request #9 from pingginp/logrus

[runtime] Use logrus and remove log.Fatal in generated table definition

vidyavnv commented:
@pingginp I just incorporated Fix #36 and did make build. But nothing really happened and CQLC. Is there any other command that I should run?

pingginp commented:
@vidyavnv hi, are you trying to use cqlc command on cassandra 3? Now (even after my fix) it only works on cassandra 2, it is swallowing errors when generating code. I will spend some time to fix the cassandra 3 issue https://github.com/pingginp/cqlc/issues/7 . A work around is run a local cassandra 2 to create the same schema and generate the code, it works on cassandra 3.

@ddave09 I am maintaining my fork but I don't have plan to actively maintain it,  I can make sure it works and fix bugs and release new tags (and go mod etc.), but I won't spend time on adding new features for it, I have to use it at work, that's the only reason I picked it up.

pingginp committed:
Fix C*3 https://github.com/pingginp/cqlc/issues/7

- `gocql.TypeText` was not mapped to string, so generated file is
invalid, could be in C*2 there is no Text in mapping ...

pingginp committed:
Merge pull request #12 from pingginp/c3

Fix C*3 https://github.com/pingginp/cqlc/issues/7

pingginp committed:
[dep] Remove logrus and use interface Fix #11

pingginp committed:
[dep] Init go mod support Fix #10

pingginp committed:
[release] 0.12.0

- support Cassandra 3
- remove logrus and use interface

pingginp committed:
[doc] Change not support C3 to support

pingginp committed:
[doc] Add go mod instruction

pingginp commented:
just an update on this PR, after some fix, the fork now supports Cassandra 3 and works w/ go mod

pingginp committed:
[gen] #7 forgot to update column type mapping

pingginp committed:
[release] 0.12.1

- previous release didn't fix the generator entirely, columnTypes
mapping also need to be updated to support text #7

pingginp committed:
[runtime] Support if in delete statement #13

pingginp committed:
[gen] Allow Eq on all columns for DELTE IF #13

- previously only primary key and those with index support it

pingginp committed:
[doc] Update change log

pingginp committed:
Merge pull request #14 from pingginp/runtime/delete-if

[runtime] Support if in delete statement #13

pingginp committed:
[release] 0.13.0
